view/RequirementsHandler.php

```diff
@@ -0,0 +1,817 @@
+	}
+
+	/**
+	 * Needed to actively prevent the inclusion of a file,
+	 * e.g. when using your own jQuery version.
+	 * Blocking should only be used as an exception, because
+	 * it is hard to trace back. You can just block items with an
+	 * ID, so make sure you add an unique identifier to customCSS() and customScript().
+	 *
+	 * @param string $fileOrID
+	 */
+	public function block($fileOrID) {
+		$this->blocked[$fileOrID] = $fileOrID;
+	}
+
+	/**
+	 * Clear either a single or all requirements.
+	 * Caution: Clearing single rules works only with customCSS and customScript if you specified a {@uniquenessID}.
+	 *
+	 * @param $file String
+	 */
+	public function clear($fileOrID = null) {
+		if($fileOrID) {
+			foreach(array('javascript','css', 'customScript', 'customCSS', 'customHeadTags') as $type) {
+				if(isset($this->{$type}[$fileOrID])) {
+					$this->disabled[$type][$fileOrID] = $this->{$type}[$fileOrID];
+					unset($this->{$type}[$fileOrID]);
+				}
+			}
+		} else {
+			$this->disabled['javascript'] = $this->javascript;
+			$this->disabled['css'] = $this->css;
+			$this->disabled['customScript'] = $this->customScript;
+			$this->disabled['customCSS'] = $this->customCSS;
+			$this->disabled['customHeadTags'] = $this->customHeadTags;
+
+			$this->javascript = array();
+			$this->css = array();
+			$this->customScript = array();
+			$this->customCSS = array();
+			$this->customHeadTags = array();
+		}
+	}
+
+	/**
+	 * Removes an item from the blocking-list.
+	 * CAUTION: Does not "re-add" any previously blocked elements.
+	 * @param string $fileOrID
+	 */
+	public function unblock($fileOrID) {
+		if(isset($this->blocked[$fileOrID])) unset($this->blocked[$fileOrID]);
+	}
+
+	/**
+	 * Removes all items from the blocking-list.
+	 */
+	public function unblockAll() {
+		$this->blocked = array();
+	}
+
+	/**
+	 * Restore requirements cleared by call to Requirements::clear
+	 */
+	public function restore() {
+		$this->javascript = $this->disabled['javascript'];
+		$this->css = $this->disabled['css'];
+		$this->customScript = $this->disabled['customScript'];
+		$this->customCSS = $this->disabled['customCSS'];
+		$this->customHeadTags = $this->disabled['customHeadTags'];
+	}
+
+	/**
+	 * Update the given HTML content with the appropriate include tags for the registered
+	 * requirements. Needs to receive a valid HTML/XHTML template in the $content parameter,
+	 * including a <head> tag. The requirements will insert before the closing <head> tag automatically.
+	 *
+	 * @todo Calculate $prefix properly
+	 *
+	 * @param string $templateFilePath Absolute path for the *.ss template file
+	 * @param string $content HTML content that has already been parsed from the $templateFilePath
+	 *                        through {@link SSViewer}.
+	 * @return string HTML content thats augumented with the requirements before the closing <head> tag.
+	 */
+	public function includeInHTML($templateFile, $content) {
```

Comment:
This code has probably changed substantially since this was originally created and we'd want to pull in the latest logic.